### PR TITLE
Add support for fuse3

### PR DESCRIPTION
**Pull request by sparlane**
Signed-off-by: Scott Parlane <scott.parlane@alliedtelesis.co.nz>

### CODE REVIEW

**Comment by sparlane** [comment]
This is currently a WIP, it builds, but I haven't tested it yet.

**Comment by brauner** [comment]
> This is currently a WIP, it builds, but I haven't tested it yet.

Thanks for working on this! Please ping me when it is in reviewable shape!

**Comment by sparlane** [comment]
@brauner 
Manual testing with it built against fuse-3.9.2 indicates it now works as expected.

File: src/bindings.c [hunk 1]
```diff
@@ -4,9 +4,17 @@
...
+
+#ifdef HAVE_FUSE3
+#ifndef FUSE_USE_VERSION
+#define FUSE_USE_VERSION 30
+#endif
+#else
 #ifndef FUSE_USE_VERSION
 #define FUSE_USE_VERSION 26
 #endif
+#endif
```
Comment (brauner) [inline src/bindings.c:17]: I wonder if we shouldn't just encapsulate this in a common header `fuse_version.h` instead of duplicating this in every file.

Comment (brauner) [reply src/bindings.c:17]: Similar to what you already did with `fuse_compat.h` below.

**Review by brauner — APPROVED** [review]
One small request for the future but overall this looks sane to me!

**Comment by brauner** [comment]
Thank you very much for this work!

**Pull request merged**